sensor/common/registry/registry_store.go

```diff
@@ -205,14 +200,19 @@ func (rs *Store) RemoveSecretID(id string) bool {
 
 // UpsertRegistry upserts the given registry with the given credentials in the given namespace into the store.
 func (rs *Store) UpsertRegistry(ctx context.Context, namespace, registry string, dce config.DockerConfigEntry) error {
-	secure, skip, err := rs.checkTLS(ctx, registry)
-	if err != nil {
-		return err
-	}
+	var err error
+	secure := true
```

Comment:
Yes, in theory no interactions with the registries should occur until after the TLS check is made and updates secure to the accurate value, however just in case we should be secure first and default to validating TLS.